File: JobRecruitment/src/main/java/bachelor/thesis/job_recruitment/core/model/Preference.java
package bachelor.thesis.job_recruitment.core.model;

import lombok.*;

import javax.persistence.*;

@EqualsAndHashCode
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Builder
public class Preference{
    @EmbeddedId
    private PreferenceKey key;

    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @MapsId("userId")
    private GenericUser user;
    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @MapsId("jobId")
    private Job job;
    private Boolean interested;
    private Boolean applied;
    private Boolean relevanceMain;
    private Boolean relevanceSec;

    @Override
    public String toString() {
        return "Preference{" +
                "key=" + key +
                ", interested=" + interested +
                ", applied=" + applied +
                ", relevanceMain=" + relevanceMain +
                ", relevanceSec=" + relevanceSec +
                '}';
    }
}

File: JobRecommender/model/Preference.py
class Preference:
    def __init__(self, user_id, job_id, interested, applied, relevanceMain, relevanceSec):
        self.user_id = user_id
        self.job_id = job_id
        self.interested = interested
        self.applied = applied
        self.relevanceMain = relevanceMain
        self.relevanceSec = relevanceSec

    def __str__(self):
        return str(self.user_id) + " " + str(self.job_id) + " " + str(self.interested) + " " + str(self.applied)
